### PR TITLE
[Keyboard] Add Shoc Keyboard

### PR DESCRIPTION
**Pull request by JannikB5825**
Pull request for a new keyboard to mainline

## Description

Keyboard and keymap definition for a new Shoc Keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
This code is incomplete and outdated. Please create the board with `qmk new-keyboard` and follow the PR checklist.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/shoc/config.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define MANUFACTURER    JannikB5825
+#define PRODUCT         SHOC
```
Comment (drashna) [inline keyboards/shoc/config.h:22]:
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/shoc/rules.mk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+MCU = atmega32u4
```
Comment (fauxpark) [inline keyboards/shoc/rules.mk:1]:
```suggestion
```

File: keyboards/shoc/info.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
+{
+    "keyboard_name": "Shoc",
```
Comment (fauxpark) [inline keyboards/shoc/info.json:2]:
```suggestion
    "keyboard_name": "Shoc",
    "manufacturer": "JannikB5825",
```

File: keyboards/shoc/info.json [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    "usb": {
+        "vid": "0xFEED",
+        "pid": "0x6060",
+        "device_version": "1.0.0"
+    },
+    "matrix_pins": {
+        "cols": ["F4", "F5", "F6", "F7", "B1", "B3", "B6", "B2"],
+        "rows": ["D3", "D2", "D4", "C6", "D7", "E6", "B4", "B5"]
+    },
+    "diode_direction": "COL2ROW",
```
Comment (fauxpark) [inline keyboards/shoc/info.json:14]:
```suggestion
    "diode_direction": "COL2ROW",
    "processor": "atmega32u4",
    "bootloader": "caterina",
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/shoc/rules.mk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
+WPM_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/shoc/rules.mk:3]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
OLED_ENABLE = yes
OLED_DRIVER = SSD1306
WPM_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**